### PR TITLE
Update Spell_monk.cpp

### PR DESCRIPTION
**Pull request by Teleslet**
fixed spell roll

[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

-  Fixed spell roll for monk
-  
-  

**Target branch(es):** 3.3.5/master

- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

	8990375
**Tests performed:** (Does it build, tested in-game, etc.)

Tested in game
**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ] 

[//]: # (**DEPRECATION NOTICE** Instead of creating new PRs to the (old) 6.x branch, make sure the target branch is **master** instead of 6.x.)
[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request (if not enabled already).)
[//]: # (**NOTE** If this PR __only__ contains SQL files, open a new issue instead and post or link the SQL there.)
[//]: # (**SUGGESTION** When adding new SQL files, name them 9999_99_99_99_db_name.sql to reduce the chance of possible merge conflicts.)

### CODE REVIEW

**Review by Keader — CHANGES_REQUESTED** [review]
Change Tabs to spaces in all file please

File: src/server/scripts/Spells/spell_monk.cpp [hunk 1]
```diff
@@ -294,11 +297,65 @@ class spell_monk_surging_mist_glyphed : public SpellScriptLoader
...
+public:
+	spell_monk_roll() : SpellScriptLoader("spell_monk_roll") { }
+
+	class spell_monk_roll_SpellScript : public SpellScript
+	{
+		PrepareSpellScript(spell_monk_roll_SpellScript);
+
+		bool Validate(SpellInfo const* /*spell*/)
+		{
+			if (!sSpellMgr->GetSpellInfo(SPELL_MONK_ROLL))
```
Comment (Keader) [inline src/server/scripts/Spells/spell_monk.cpp]: you should validate here ```SPELL_MONK_ROLL_TRIGGER```

File: src/server/scripts/Spells/spell_monk.cpp [hunk 2]
```diff
@@ -294,11 +297,65 @@ class spell_monk_surging_mist_glyphed : public SpellScriptLoader
...
+			if (!app)
+				return;
+
+			app->ClientUpdate();
+		}
+
+		void HandleAfterCast()
+		{
+			Unit* caster = GetCaster();
+			if (!caster || caster->GetTypeId() != TYPEID_PLAYER)
```
Comment (Keader) [inline src/server/scripts/Spells/spell_monk.cpp]: Caster cant be null in SpellScript, and this check ```caster->GetTypeId() != TYPEID_PLAYER``` should be in Load function :)

File: src/server/scripts/Spells/spell_monk.cpp [hunk 3]
```diff
@@ -294,11 +297,65 @@ class spell_monk_surging_mist_glyphed : public SpellScriptLoader
...
+				return;
+
+			caster->CastSpell(caster, SPELL_MONK_ROLL_TRIGGER, true);
+
+		}
+
+		void Register()
```
Comment (Keader) [inline src/server/scripts/Spells/spell_monk.cpp]: missing override

File: src/server/scripts/Spells/spell_monk.cpp [hunk 4]
```diff
@@ -294,11 +297,65 @@ class spell_monk_surging_mist_glyphed : public SpellScriptLoader
...
+		{
+			BeforeCast += SpellCastFn(spell_monk_roll_SpellScript::HandleBeforeCast);
+			AfterCast += SpellCastFn(spell_monk_roll_SpellScript::HandleAfterCast);
+		}
+	};
+
+	SpellScript* GetSpellScript() const
```
Comment (Keader) [inline src/server/scripts/Spells/spell_monk.cpp]: missing override

File: src/server/scripts/Spells/spell_monk.cpp [hunk 5]
```diff
@@ -307,10 +307,12 @@ class spell_monk_roll : public SpellScriptLoader
 	{
 		PrepareSpellScript(spell_monk_roll_SpellScript);
 
-		bool Validate(SpellInfo const* /*spell*/)
+		bool Validate(SpellInfo const* /*spell*/) override
 		{
 			if (!sSpellMgr->GetSpellInfo(SPELL_MONK_ROLL))
```
Comment (Keader) [inline src/server/scripts/Spells/spell_monk.cpp]: you just need check spells used inside of your spellscript.
in this case, only ```SPELL_MONK_ROLL_TRIGGER```

File: src/server/scripts/Spells/spell_monk.cpp [hunk 6]
```diff
@@ -330,21 +332,17 @@ class spell_monk_roll : public SpellScriptLoader
 		void HandleAfterCast()
 		{
 			Unit* caster = GetCaster();
-			if (!caster || caster->GetTypeId() != TYPEID_PLAYER)
```
Comment (Keader) [inline src/server/scripts/Spells/spell_monk.cpp]: add it:

```cpp
            bool Load() override
            {
                return GetCaster()->GetTypeId() == TYPEID_PLAYER;
            }
```
below or above Validate function.

**Comment by Keader** [comment]
Missing a sql file
something like:
```sql

DELETE FROM `spell_script_names` WHERE `ScriptName`='spell_monk_roll';
INSERT INTO `spell_script_names` (`spell_id`, `ScriptName`) VALUES
(109132,'spell_monk_roll');

```

**Comment by Keader** [comment]
Any news on this @Teleslet?
Still missing sql file and still missing change tabs to 4 spaces

**Comment by Keldo** [comment]
Still doesnt work, Click, Sound Effects Work, No Action, Timer is present --- No Roll.

**Comment by Shauren** [comment]
PR doesn't work at all? K.

**Pull request closed without merging**